scene/3d/physics_body.cpp

```diff
@@ -92,17 +136,42 @@ void PhysicsBody::remove_collision_exception_with(Node* p_node) {
 	PhysicsServer::get_singleton()->body_remove_collision_exception(get_rid(),physics_body->get_rid());
 }
 
+void PhysicsBody::_set_layers(uint32_t p_mask) {
```

Comment:
For 3D stuff collision exclusion/inclusion is more intuitive to use than
layers. It would be better to add that for all colsiion bodies (move
esisting stuff to the root of hierarchy)
 instead of layers thingy. Really.


On Tue, Apr 12, 2016 at 4:09 PM, Rémi Verschelde <notifications@github.com>
wrote:

> In scene/3d/physics_body.cpp
> <https://github.com/godotengine/godot/pull/4273#discussion_r59372123>:
>
> > @@ -92,17 +136,42 @@ void PhysicsBody::remove_collision_exception_with(Node* p_node) {
> >  	PhysicsServer::get_singleton()->body_remove_collision_exception(get_rid(),physics_body->get_rid());
> >  }
> >
> > +void PhysicsBody::_set_layers(uint32_t p_mask) {
>
> Why isn't is shown in the class ref? And that's precisely because it works
> for set("layers", value) that a deprecation warning could be useful, so
> that developers using this know that they should switch to
> set_layer_mask(value) and set_collision_mask(value) ;)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/4273/files/f7c3d6329cdc9c21c205f1522813090313a422d2#r59372123>
>